students/hyska_monika/lesson_09_reading_and_writing_files/regex_search.py

```diff
@@ -0,0 +1,37 @@
+"""
+Program that opens all .txt files in a folder and
+searches for any line that matches a user-supplied regular expression.
+The results should be printed to the screen.
+"""
+
+from os import listdir
+from os.path import isfile, join, os
+import re
+
+
+def files_in_path(path):
+    only_files = [f for f in listdir(path) if isfile(join(path, f))]
```

Comment:
```
import os
...
... os.listdir(path)...
```
is recommended, importing names directly into module is discouraged. The reason for that is that the names will leak into any module that imports this one. Also `os.path.*` is accessible after `os` is imported (some magic).